src/compas/datastructures/halfedge/halfedge.py

```diff
@@ -2092,76 +2092,75 @@ def edge_loop(self, edge):
         list of tuple of int
             The edges on the same loop as the given edge.
         """
-        if self.is_edge_on_boundary(*edge):
-            return self._edge_loop_on_boundary(edge)
-        edges = []
-        current, previous = edge
-        edges.append((previous, current))
+        u, v = edge
+        uv_loop = self.halfedge_loop((u, v))
+        if uv_loop[0][0] == uv_loop[-1][1]:
+            return uv_loop
+        vu_loop = self.halfedge_loop((v, u))
+        vu_loop[:] = [(u, v) for v, u in vu_loop[::-1]]
+        return vu_loop + uv_loop[1:]
```

Comment:
for example, i think the test exits with the first `if` and the code afterward isn't covered.